bindings/tflite/java/org/tensorflow/lite/native/interpreter_jni.cc

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <jni.h>
+
+#include "iree/base/logging.h"
+
+// NOTE: we pull in our own copy here in case the tflite API changes upstream.
+#define TFL_COMPILE_LIBRARY 1
+#include "bindings/tflite/include/tensorflow/lite/c/c_api.h"
+
+#define JNI_FUNC extern "C" JNIEXPORT
+#define JNI_PREFIX(METHOD) Java_org_tensorflow_lite_Interpreter_##METHOD
+
+namespace {
+
+// Returns a pointer to the native IREE module stored by the GetInterpreter
+// object.
+static TfLiteInterpreter* GetInterpreter(JNIEnv* env, jobject obj) {
+  jclass clazz = env->GetObjectClass(obj);
+  IREE_DCHECK(clazz);
+
+  jfieldID field = env->GetFieldID(clazz, "nativeAddress", "J");
```

Comment:
GetFieldID can throw - you'll need to check if env->ExceptionCheck() is true and if so return NULL. In JNI it's unsafe to call any JNI method (including most on JNIEnv) while an exception is pending so this unfortunately is not optional.

So you want:
```c++
JNI_FUNC jint JNI_PREFIX(nativeAllocateTensors)(JNIEnv* env, jobject thiz) {
  TfLiteInterpreter* interpreter = GetInterpreter(env, thiz);
  if (!interpreter) return 0;  // default value; exception is set and java will handle it upon return
  ...
}
```

Reply:
I've updated nativeAllocateTensors, wondering if I need to propagate this check to every method?